rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -122,11 +125,11 @@ __rmw_publisher_get_actual_qos(
   rmw_qos_profile_t * qos)
 {
   auto info = static_cast<CustomPublisherInfo *>(publisher->data);
-  eprosima::fastrtps::Publisher * fastrtps_pub = info->publisher_;
-  const eprosima::fastrtps::PublisherAttributes & attributes =
-    fastrtps_pub->getAttributes();
+  eprosima::fastdds::dds::DataWriter * fastrtps_pub = info->publisher_;
```

Comment:
```suggestion
  eprosima::fastdds::dds::DataWriter * fastdds_dw = info->publisher_;
```

I would advise to avoid any use of fastrtps when possible. In this case, it is an internal variable so there should not be any problem changing its name.